src/new/wallet.js

```diff
@@ -2322,6 +2304,111 @@ class HathorWallet extends EventEmitter {
     const tx = await this.prepareCreateNewToken(name, symbol, amount, newOptions);
     return this.handleSendPreparedTransaction(tx);
   }
+
+  /**
+   * Identify all inputs from the loaded wallet
+   *
+   * @param {Transaction} tx The transaction
+   *
+   * @returns {{
+   * inputIndex: number,
+   * addressIndex: number,
+   * addressPath: string,
+   * }[]} List of indices and their associated address index
+   */
+   getWalletInputInfo(tx) {
+    storage.setStore(this.store);
+
+    const walletInputs = [];
+
+    for (const [inputIndex, input] of tx.inputs.entries()) {
+      const inputTx = this.getTx(input.hash);
+      if (!inputTx) {
+        // Input is not from our wallet
+        continue;
+      }
+      const utxo = inputTx.outputs[input.index];
+      if (utxo && this.isAddressMine(utxo.decoded.address)) {
+        const addressIndex = this.getAddressIndex(utxo.decoded.address);
+        // BIP32 address path
+        const addressPath = this.getAddressPathForIndex(addressIndex);
+        walletInputs.push({ inputIndex, addressIndex, addressPath });
+      }
+    }
+
+    return walletInputs;
+  }
+
+  /**
+   * Get signatures for all inputs of the loaded wallet.
```

Comment:
No, since it returns the ECDSA signature with some metadata (e.g. address index, pubkey) it can be used by both P2PKH and P2SH.
I've also changed the `getAllSignatures` which is P2SH only to use this method.